### PR TITLE
Move to IJ 2016 as the lowest supported version of IJ

### PR DESCRIPTION
**Pull request by patflynn**
includes setting the project language level to java 8 since 2016 is java 8 based.

I had to upgrade the gradle-intellij-plugin to get this to build with the new config.

the random test case refactor is error-prone suddenly behaving slightly differently with the config change.

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -13,17 +13,22 @@
...
+      <module name="common-test-lib_test" target="1.8" />
+      <module name="google-account-plugin_main" target="1.8" />
+      <module name="google-account-plugin_test" target="1.8" />
+      <module name="google-cloud-tools-plugin_main" target="1.8" />
+      <module name="google-cloud-tools-plugin_test" target="1.8" />
+      <module name="ultimate_main" target="1.8" />
+      <module name="ultimate_test" target="1.8" />
     </bytecodeTargetLevel>
   </component>
 </project>
```
Comment (joaoandremartins) [inline .idea/compiler.xml]: Missing eol

**Comment by etanshaul** [comment]
I'm happy to see that upgrading gradle-intellij-plugin didn't cause you any trouble with the nested module structure of our project. This was why I delayed the upgrade prevously.

**Comment by patflynn** [comment]
Hopefully it hasn't. Still doing some manual testing.

File: google-account-plugin/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -13,15 +13,13 @@
...
+
 <idea-plugin version="2">
   <id>com.google.gct.login</id>
   <name>Google Account</name>
   <vendor>Google</vendor>
   <!-- "version" set by gradle-intellij-plugin -->
   <!-- "idea-version since-build" set by gradle-intellij-plugin -->
-  <!-- Workaround for gradle-intellij-plugin limitation: https://github.com/JetBrains/gradle-intellij-plugin/issues/66 -->
-  <!-- Setting version to 15.0.6 and up. https://confluence.jetbrains.com/display/IDEADEV/IDEA+15+EAP -->
-  <idea-version since-build="143.2370.31" />
```
Comment (etanshaul) [inline google-account-plugin/resources/META-INF/plugin.xml]: don't we still want to set a version here so that it shows up properly in the IJ repo page?

Comment (etanshaul) [inline google-account-plugin/resources/META-INF/plugin.xml]: See https://github.com/JetBrains/gradle-intellij-plugin/issues/66 for the new way to specify version ranges.

Comment (patflynn) [inline google-account-plugin/resources/META-INF/plugin.xml]: by default it's configured to the version we are running the build against.  Although I'll double-check

**Comment by patflynn** [comment]
Yeah I saw that,  but it's only available in a snapshot release.

**Comment by patflynn** [comment]
@etanshaul PTAL

**Comment by etanshaul** [comment]
The only thing I see remaining is my comment about being explicit about our supported versions: e.g. specifying 2016+ in plugin.xml. Otherwise LGTM.

**Comment by patflynn** [comment]
I'll reintroduce that as soon as we have a nice way of specifying it.  Right now I'm loathe to lookup the right build number every time we update the sdk version we build against.

**Comment by etanshaul** [comment]
sgtm if you confirmed that it implicitly will be set to 2016+ if nothing is specified.

**Comment by patflynn** [comment]
Thanks for pushing on that. I checked and it wasn't. pushed a change that
fixes that.

**Pull request merged**